### ISSUE TITLE
Autorest 'public-clients : true' doesn't generate public clients 

### ISSUE THREAD

**Issue by swashtek** [Azure/autorest.csharp#1210]
Autorest (3.2.1) with 'public-clients: true' flag doesn't generate public clients. We don't see constructor which takes parameters for http client or base uri. Also, there is another constructor with ClientDiagnostics and HttpPipeline which is internal, so not accessible. 
How do we get around this ? Are there any steps to perform to make it accessible ? 
Please direct us to examples if any. Thank you. 

``` yaml
clear-output-folder: true
input-file: swagger.json
output-folder: Generated 
csharp: true
namespace: SampleNamespace
override-client-name: SampleClient
v3: true
verbose: true
skip-csproj: true
public-clients : true
```

**Comment by chamons** [comment]
This sample does use public-clients successfully:

https://github.com/Azure/autorest.csharp/blob/feature/v3/samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs#L37

Please post a full example project so I can try to reproduce your issue.

**Comment by davidkaya** [comment]
@chamons I actually also encountered this issue just today. I've created a small repro project: https://github.com/davidkaya/public-clients-repro

I've tried to set `public-clients` also through CLI, didn't help.

```
❯ autorest --info
AutoRest code generation utility [cli version: 3.2.1; node: v14.16.1, max-memory: 4096 MB]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest


Showing All Installed Extensions

 Type       Extension Name                           Version      Location
 core       @autorest/core                           3.1.3        /Users/davidkaya/.autorest/@autorest_core@3.1.3
 core       @autorest/core                           3.4.2        /Users/davidkaya/.autorest/@autorest_core@3.4.2
 extension  @autorest/csharp                         3.0.0-beta.20210527.2 /Users/davidkaya/.autorest/@autorest_csharp@3.0.0-beta.20210527.2
 extension  @autorest/modelerfour                    4.19.1       /Users/davidkaya/.autorest/@autorest_modelerfour@4.19.1
```

**Comment by chamons** [comment]
I suspect you've hit https://github.com/Azure/autorest.csharp/issues/1146

Try changing:

```
csharp:
  public-clients: true
```

to

```
csharpgen:
  public-clients: true
```

**Comment by kamserv45** [comment]
Running into this same issue. Tried changing `csharp` to `csharpgen` with no fix. In the configuration.yaml that gets generated, I can see public-clients=true in the csharpgen section as well as just outside of that section because I've tried to have that flag in both places. It must in some capacity recognize that flag because when I set public-clients to be false, the APIClient class doesn't show up at all, but the APIRestClient does. Like the OP said, with that flag set to true, the APIClient class is generated public, but the constructors are unusable as the constructor with ClientDiagnostics, HttpPipeline is marked as internal.
